tests/SimplifyTest.elm

```diff
@@ -19373,6 +19374,208 @@ a = Task.succeed << f
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Task.map3 where each task is a succeeding task will result in Task.succeed on the values inside"
+                            , details = [ "You can replace this call by Task.succeed with the function applied to the values inside each succeeding task." ]
+                            , under = "Task.map3"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Task
+a = (c |> g) |> f a b |> Task.succeed
+"""
+                        ]
+        , test "should replace Task.map3 f (Task.succeed a) (Task.fail x) task2 by (Task.fail x)" <|
+            \() ->
+                """module A exposing (..)
+import Task
+a = Task.map3 f (Task.succeed a) (Task.fail x) task2
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Task.map3 where we know the first failing task will result in that failing task"
+                            , details = [ "You can replace this call by the first failing task." ]
+                            , under = "Task.map3"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Task
+a = (Task.fail x)
+"""
+                        ]
+        , test "should replace Task.map3 f (Task.succeed a) (Task.fail x) by always (Task.fail x)" <|
+            \() ->
+                """module A exposing (..)
+import Task
+a = Task.map3 f (Task.succeed a) (Task.fail x)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Task.map3 where we know the first failing task will result in that failing task"
+                            , details = [ "You can replace this call by always with the first failing task." ]
+                            , under = "Task.map3"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Task
+a = always (Task.fail x)
+"""
+                        ]
+        , test "should replace Task.map3 f (Task.fail x) task1 task2 by (Task.fail x)" <|
+            \() ->
+                """module A exposing (..)
+import Task
+a = Task.map3 f (Task.fail x) task1 task2
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Task.map3 where we know the first failing task will result in that failing task"
+                            , details = [ "You can replace this call by the first failing task." ]
```

Comment:
I just noticed that the wording is different if the first element is failing than if it's the second one, and I'm not sure why the first argument is special here :thinking:

Reply:
Because they are different situations: One is for when we only know that arguments after are irrelevant because tasks earlier could fail, too. The other replaces the whole expression by the first _known_ failing task.

Or am I missing something?

Reply:
No that clears it up, thanks 👍